src/main/java/com/codeborne/selenide/logevents/SimpleReport.java

```diff
@@ -72,16 +72,26 @@ String generateReport(String title, List<LogEvent> events) {
     return report.build();
   }
 
-  private List<LogEventWithNestingLevel> computeNestingLevel(List<LogEvent> events) {
-    var stack = new ArrayDeque<LogEvent>();
+  private List<LogEventWithNestingLevel> escapeSubjectAndComputeNestingLevel(List<LogEvent> events) {
+    var stack = new ArrayDeque<LogEventWithNestingLevel.SimpleLogEvent>();
     var result = new ArrayList<LogEventWithNestingLevel>(events.size());
     for (LogEvent event : events) {
-      while (!stack.isEmpty() && stack.peekFirst().getEndTime() <= event.getStartTime())
+      var eventWithEscapedSubject = new LogEventWithNestingLevel.SimpleLogEvent(
+        event.getElement(),
+        escape(event.getSubject()),
```

Comment:
At a first glance, the good moment to escape characters is not the construction of `SimpleLogEvent`, but when we render `SimpleLogEvent`. In other words, when we generate the resulting report. Then we could escape all fields, not only subject.

Reply:
But we need to know the max size of a cell of the report table. If the calculation was made before the `escape` was applied then the final max size may be incorrect.

Reply:
Since `element` and `subject` are just strings, I believe, that they are  the only place where special characters can appear. So, it's enough to escape them, not all the report line/row.

But, I also didn't find a case when the `element` could contain some special characters (only some synthetic example), that's why I left it as is, but it can be easily fixed.